tidy3d/components/data/monitor_data_n2f.py

```diff
@@ -36,24 +90,77 @@ def normalize(self, source_spectrum_fn: Callable[[float], complex]) -> AbstractN
 
         return self.copy(update=fields_norm)
 
-    @staticmethod
-    def nk(frequency: float, medium: Medium) -> Tuple[float, float]:
+    def nk(self, medium: Medium) -> Tuple[float, float]:
```

Comment:
If we end up using a `data.py` strategy that retains access to the associated `monitor`, maybe I should just get the `medium` from the `AbstractNear2FarMonitor`. I realize now that there's a redundancy - also a point Xin brought up for the GUI.

Reply:
yea, I guess the only reason to expose medium in the kwargs is if the user might want to project with a medium separate than that in the monitor. So if we don't care about that and we end up refactoring in a way that the medium is still available through the monitor, it might not be needed.  Alternatively, we could make these generic and replace `medium: Medium` with `propagation_constant: complex` and then feed the medium. Let's wait to do anything else on this point until we figure out the reorganization.